### PR TITLE
[core] Check if model file exists on read model

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Add more detailed error messages when model and/or weights files not exist at path

### Tickets:
 - [CVS-142133](https://jira.devtools.intel.com/browse/CVS-142133)

### CODE REVIEW

File: src/inference/tests/unit/core.cpp [hunk 1]
```diff
@@ -407,6 +406,37 @@ TEST(CoreTests_parse_device_config, get_batch_device_name) {
...
 
+TEST(CoreTests_read_model, model_not_exist_at_path) {
+    ov::Core core;
+
+    auto model_file_path = ov::test::utils::generateTestFilePrefix();
+    model_file_path += "not_existing_model.xml";
+
+    OV_EXPECT_THROW(core.read_model(model_file_path),
+                    ov::Exception,
+                    testing::HasSubstr("Model file not exists at: " + model_file_path));
```
Comment (t-jankowski) [inline src/inference/tests/unit/core.cpp:417]:
```suggestion
                    testing::HasSubstr("Model file does not exist at: " + model_file_path));
```

File: src/inference/tests/unit/core.cpp [hunk 2]
```diff
@@ -407,6 +406,37 @@ TEST(CoreTests_parse_device_config, get_batch_device_name) {
...
+
+    {
+        std::ofstream model_file;
+        model_file.open(model_file_path);
+        model_file.close();
+    }
+
+    OV_EXPECT_THROW(core.read_model(model_file_path, weights_file_path),
+                    ov::Exception,
+                    testing::HasSubstr("Model's weights file not exists at: " + weights_file_path));
```
Comment (t-jankowski) [inline src/inference/tests/unit/core.cpp:435]:
```suggestion
                    testing::HasSubstr("Model's weights file does not exist at: " + weights_file_path));
```

File: src/inference/src/model_reader.cpp [hunk 3]
```diff
@@ -126,6 +128,7 @@ std::shared_ptr<ov::Model> read_model(const std::string& modelPath,
     ov::AnyVector params{model_path};
 
     if (!binPath.empty()) {
+        OPENVINO_ASSERT(file_exists(binPath.c_str()), "Model's weights file not exists at: ", binPath);
```
Comment (t-jankowski) [inline src/inference/src/model_reader.cpp:131]:
```suggestion
        OPENVINO_ASSERT(file_exists(binPath.c_str()), "Model's weights file does not exist at: ", binPath);
```

File: src/inference/src/model_reader.cpp [hunk 4]
```diff
@@ -111,6 +111,8 @@ std::shared_ptr<ov::Model> read_model(const std::string& modelPath,
                                       const std::string& binPath,
                                       const std::vector<ov::Extension::Ptr>& extensions,
                                       bool enable_mmap) {
+    OPENVINO_ASSERT(file_exists(modelPath.c_str()), "Model file not exists at: ", modelPath);
```
Comment (t-jankowski) [inline src/inference/src/model_reader.cpp:114]:
```suggestion
    OPENVINO_ASSERT(file_exists(modelPath.c_str()), "Model file does not exist at: ", modelPath);
```

**Review by t-jankowski — CHANGES_REQUESTED** [review]

File: src/inference/src/model_reader.cpp [hunk 5]
```diff
@@ -111,6 +111,8 @@ std::shared_ptr<ov::Model> read_model(const std::string& modelPath,
                                       const std::string& binPath,
                                       const std::vector<ov::Extension::Ptr>& extensions,
                                       bool enable_mmap) {
+    OPENVINO_ASSERT(file_exists(modelPath.c_str()), "Model file does not exist at: ", modelPath);
```
Comment (ilya-lavrenov) [inline src/inference/src/model_reader.cpp:114]: what if frontend is called directly? should we move it somewhere to base frontend class?

Comment (praasz) [reply src/inference/src/model_reader.cpp]: Move check into frontends. As each have little different logic update base class not always works correctly.

File: src/frontends/common/src/frontend.cpp [hunk 6]
```diff
@@ -108,3 +108,19 @@ std::string FrontEnd::get_name() const {
...
+
+void FrontEnd::validate_file_stream(const std::ifstream& stream, const std::string& model_path) {
+    if (!stream.is_open()) {
+        throw_path_error(model_path);
+    }
+}
+
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+void FrontEnd::validate_file_stream(const std::ifstream& stream, const std::wstring& model_paths) {
+    validate_file_stream(ov::util::wstring_to_string(model_path), stream);
```
Comment (t-jankowski) [inline src/frontends/common/src/frontend.cpp:124]:
```suggestion
    validate_file_stream(stream, ov::util::wstring_to_string(model_path));
```

File: src/frontends/tensorflow/src/frontend.cpp [hunk 7]
```diff
@@ -178,6 +182,8 @@ bool FrontEnd::supported_impl(const std::vector<ov::Any>& variants) const {
         } else if (GraphIteratorProtoTxt::is_supported(model_path)) {
             // handle text protobuf format
             return true;
+        } else {
+            throw_path_error(ov::util::wstring_to_string(model_path));
```
Comment (t-jankowski) [inline src/frontends/tensorflow/src/frontend.cpp:186]: What about overriding `throw_path_error` to accept also `wstring`?

File: src/frontends/common/src/frontend.cpp [hunk 8]
```diff
@@ -108,3 +108,19 @@ std::string FrontEnd::get_name() const {
     }
     FRONTEND_RETURN_STATEMENT("Getting frontend name", m_actual->get_name();)
 }
+
+void FrontEnd::throw_path_error(const std::string& model_path) {
+    FRONT_END_GENERAL_CHECK(false, "Could not open the file: ", model_path);
```
Comment (ilya-lavrenov) [inline src/frontends/common/src/frontend.cpp:113]: can we escape model_path with quotes? 
I saw when people passed extra spaces to model_path, so we can highlight it with such quotes

File: src/frontends/tensorflow/src/frontend.cpp (hunk 7 above)
Comment (praasz) [reply src/frontends/tensorflow/src/frontend.cpp]: Added

**Review by t-jankowski — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**